ngrok.go

```diff
@@ -71,15 +176,21 @@ func (*Ngrok) CaddyModule() caddy.ModuleInfo {
 
 // WrapListener return an ngrok listener instead the listener passed by Caddy
 func (n *Ngrok) WrapListener(net.Listener) net.Listener {
+	if err := n.ProvisionOpts(); err != nil {
+		panic(err)
+	}
```

Comment:
I had read this in the caddy docs 

"Additionally, you should avoid performing expensive operations in Provision, since provisioning is performed even if a config is only being validated. When in the provisioning phase, do not expect that the module will actually be used."

So I moved the actual options provisioning to the function that runs when we're actually going to use the tunnel. I had meant to make this function additionally return an error and check it.

Do you have thoughts on this?

Reply:
The documentation speaks of expensive operations as-in making a network call. What we're doing in `provisionOpts` isn't expensive, and it's best if we return an error early than panic.